packages/lodestar/src/chain/chain.ts

```diff
@@ -139,13 +139,9 @@ export class BeaconChain implements IBeaconChain {
   public async getHeadStateContext(): Promise<ITreeStateContext> {
     // head state should always exist
     const head = this.forkChoice.getHead();
-    const headStateRoot =
-      (await this.db.checkpointStateCache.getLatest({
-        root: head.blockRoot,
-        epoch: Infinity,
-      })) || (await this.regen.getState(head.stateRoot));
-    if (!headStateRoot) throw Error("headStateRoot does not exist");
-    return headStateRoot;
+    const headState = await this.db.stateCache.get(head.stateRoot);
```

Comment:
this should keep the same logic to get from `checkpointStateCache` then `stateCache`, otherwise we can't start from an anchor state at orphaned block